pkg/document/crdt/element_rht.go

```diff
@@ -129,17 +129,17 @@ func (rht *ElementRHT) Delete(k string, deletedAt *time.Ticket) Element {
 }
 
 // DeleteByCreatedAt deletes the Element of the given creation time.
-func (rht *ElementRHT) DeleteByCreatedAt(createdAt *time.Ticket, deletedAt *time.Ticket) Element {
+func (rht *ElementRHT) DeleteByCreatedAt(createdAt *time.Ticket, deletedAt *time.Ticket) (Element, error) {
 	node, ok := rht.nodeMapByCreatedAt[createdAt.Key()]
 	if !ok {
-		return nil
+		return nil, fmt.Errorf("fail to find the given createdAt: %s", createdAt.Key())
```

Comment:
It is better to keep the context succulent by avoiding "failed to".
https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

Reply:
I guess this hasn't been fixed yet.